atosl.c

```diff
@@ -882,45 +882,29 @@ void dwarf_mach_object_access_finish(Dwarf_Obj_Access_Interface *obj)
     free(obj);
 }
 
-const char *lookup_symbol_name(Dwarf_Addr addr)
+struct dwarf_subprogram_t *lookup_symbol(Dwarf_Addr addr)
 {
     struct dwarf_subprogram_t *subprogram = context.subprograms;
 
     while (subprogram) {
         if ((addr >= subprogram->lowpc) &&
-            (addr <= subprogram->highpc)) {
-            return subprogram->name;
-            break;
+            (addr < subprogram->highpc)) {
```

Comment:
Yeah. And really I should make that a separate diff to make it obvious. I think I'll do that.